SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -595,14 +566,65 @@ void UGlobalStateManager::BecomeAuthoritativeOverAllActors()
 	}
 }
 
+void UGlobalStateManager::BecomeAuthoritativeOverActorsBasedOnLBStrategy()
+{
+	for (TActorIterator<AActor> It(NetDriver->World); It; ++It)
+	{
+		AActor* Actor = *It;
+		if (Actor != nullptr && !Actor->IsPendingKill())
+		{
+			if (Actor->GetIsReplicated() && NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*Actor))
```

Comment:
We should probably make a similar change to this in SpatialNetDriver::OnLevelAddedToWorld, which handles sublevels (currently the worker authoritative over the GSM will take authority over streamed in actors).  It's fine if it comes in a subsequent PR, just wanted to mention it.